### PR TITLE
Administrators page form updates

### PR DESCRIPTION
**Pull request by kelbyhawn**
Refactor the Salesforce form code on https://code.org/administrators to:
- Put the Javascript code in it's own file in the `pegasus/sites.v3/code.org/public/js` directory
- Show a disabled state on the Submit button until the reCAPTCHA checkbox is checked (see video)

Tested by submitting the form and checking w/ the team to see if the response was submitted in Salesforce. 

**Relevant links:** [Asana task](https://app.asana.com/0/0/1203947967421592) • Related PR: https://github.com/code-dot-org/code-dot-org/pull/50079

https://user-images.githubusercontent.com/9256643/218832188-17f66285-2524-47cd-8915-db345de1bdde.mov

### CODE REVIEW

File: pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js
Comment (kelbyhawn) [inline pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js]: Uses the `data-callback` attribute [here](https://github.com/code-dot-org/code-dot-org/blob/dd288f1b07cbbeefe1aad70495117cbfe9a35e82/pegasus/sites.v3/code.org/views/admins_email_signup_form.html#L44) to check for a successful response — https://developers.google.com/recaptcha/docs/display

File: pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js
Comment (megcrenshaw) [inline pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js:1]: Could you link the actual site? I have a feeling we'll want to reference it often as none of us are familiar with this

**Review by TurnerRiley — APPROVED** [review]
This looks great! I agree with Meg's comment about including the actual site in this PR if it isn't too urgent to get out.

File: pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js
Comment (kelbyhawn) [reply pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js:1]: Yes definitely! Alek gave me this code though and said she got it from you, so I'm not sure what to link 😅

Comment (megcrenshaw) [reply pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js:1]: Ha, let's see what I can find ...

Comment (megcrenshaw) [reply pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js:1]: I really think Alek found this, but I had it in my history!
https://www.sfdcpoint.com/salesforce/salesforce-web-to-lead-form-with-recaptcha/

Looks like this is coming from a blog about Salesforce, which I hope is credible enough 🙈

File: pegasus/sites.v3/code.org/views/admins_email_signup_form.html
Comment (megcrenshaw) [inline pegasus/sites.v3/code.org/views/admins_email_signup_form.html:2]: Thank you for extracting this out!

File: pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js
Comment (kelbyhawn) [reply pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js:1]: Haha I think that works! 😂 Adding now…

File: pegasus/sites.v3/code.org/views/admins_email_signup_form.html
Comment (megcrenshaw) [inline pegasus/sites.v3/code.org/views/admins_email_signup_form.html:44]: After talking with Dave, we might want to figure out a way to make this sitekey harder to access. I think we're safe for now because we're not allowing `studio.code.org` access to the recaptcha (for now), but wanted to flag it here –– I'll write up a doc tomorrow so that this is tracked.

File: pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js
Comment (megcrenshaw) [inline pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js]: I'm confused about this –– we remove the attribute and then set it too?

File: pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js
Comment (megcrenshaw) [inline pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js]: Does this need to be called here, or is referencing it in the HTML enough?

File: pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js
Comment (kelbyhawn) [reply pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js:1]: Added!

File: pegasus/sites.v3/code.org/views/admins_email_signup_form.html
Comment (megcrenshaw) [reply pegasus/sites.v3/code.org/views/admins_email_signup_form.html:44]: Could you link https://developers.google.com/recaptcha/docs/display as well for reference? It's looking like the `class="g-recaptcha"` tag is doing some important stuff and I don't want the details to get lost

File: pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js
Comment (kelbyhawn) [reply pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js]: I was getting a linting error when I tried committing w/o it.

Comment (megcrenshaw) [reply pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js]: My hunch is that we don't need to run the function here –– it should get executed at the right time via the HTML `data-callback`

File: pegasus/sites.v3/code.org/views/admins_email_signup_form.html
Comment (megcrenshaw) [reply pegasus/sites.v3/code.org/views/admins_email_signup_form.html:44]: Great job using the `data-callback` functionality for validation –– this is a very clean solution imo!

File: pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js
Comment (kelbyhawn) [reply pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js]: Oh yeah, we don't need to set it too! I adapted this from [this Stackoverflow answer](https://stackoverflow.com/a/40881272) and did a 1:1 swap w/ the attribute methods bc my JS skills are quite rusty 😅 🤦🏻‍♀️ I'll remove the .setAttribute() method now!

File: pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js
Comment (kelbyhawn) [reply pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js]: Removed!

**Comment by davidsbailey** [comment]
great work Kelby and thanks for getting into the weeds Meg! It would be great if you can write down your plan for how to test this, both that it works with captcha answered and doesn't work without -- I don't think captcha works on localhost, but will you at least be able to test it on staging before we ship this to production?

I'm not sure if there might be security implications with allowlisting localhost-studio.code.org for recaptcha, but @megcrenshaw  , that would be a good option to discuss in the doc you are talking about writing up. when you do, I'll add my thoughts about security implications there.

File: pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js
Comment (megcrenshaw) [reply pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js]: Ah, sorry, I missed your comment before I wrote mine above! Sounds great –– thank you!

File: pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js
Comment (megcrenshaw) [inline pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js]: This looks like it's wrapped weirdly which makes it hard to parse. Not blocking, but could you add line breaks and indentation here?

**Review by megcrenshaw — APPROVED** [review]
Well done, Kelby! Love the callback solution for validation.

I think there's some follow-up work on the engineering side that I can take for now –– will loop you in on those conversations!

File: pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js
Comment (megcrenshaw) [reply pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js]: I'm getting lint errors here –– not sure this will pass drone?

Comment (davidsbailey) [reply pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js]: @kelbyhawn to continue from our previous discussion, these lint errors are part of what we gain by extracting the js 🎉  apologies that they are extra work (especially when they just complain about formatting), but the benefit is that they can catch typos and other errors in our js

Comment (davidsbailey) [reply pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js]: to make sure you see the lint errors in your local dev environment, make sure you have installed the "hooks" as described in https://github.com/code-dot-org/code-dot-org/blob/staging/SETUP.md#overview

Comment (megcrenshaw) [reply pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js]: Thanks for adding this context, Dave! Forgot, Kelby, you may not have linting set up –– reach out if you have questions about this.

Comment (kelbyhawn) [reply pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js]: Thanks @davidsbailey and @megcrenshaw! I do have linting set up 🙌 

I just reformatted w/ the line breaks, so the only error I'm getting now is `21:10  error  'recaptchaCallback' is defined but never used  no-unused-vars`, which is related to this comment: https://github.com/code-dot-org/code-dot-org/pull/50272#discussion_r1106297968 — any thoughts on what I should do here?

Comment (megcrenshaw) [reply pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js]: Ah yes. Gnarly. Our linter doesn't pick up that it's being used in an HTML file!

So you can do this
```
// eslint-disable-next-line no-unused-vars
function recaptchaCallback() {
```

Simpler would be
```
function recaptchaCallback() { // eslint-disable-line no-unused-vars
```
but then it gets mad about line length being too long. What a time to be alive.

But at least we're still specific about (a) the error to disable (no-unused-vars) and (b) the exact line

Comment (kelbyhawn) [reply pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js]: That worked, thanks! Just updated 🎉 

(I lol'd at "What a time to be alive." 😂)

**Comment by kelbyhawn** [comment]
> Well done, Kelby! Love the callback solution for validation.
> 
> I think there's some follow-up work on the engineering side that I can take for now –– will loop you in on those conversations!

Woo hoo, thank you! Do you want to do another review since I made a few updates to address comments, or should I be ok to merge once the tests pass?

File: pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js
Comment (davidsbailey) [reply pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js:1]: ditto 😂 

either eslint-disable solution seems fine to me.

File: pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js
Comment (davidsbailey) [inline pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js:15]: thank you for extracting this, it is much easier to read now!

File: pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js
Comment (davidsbailey) [inline pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js:28]: one thing to note is that the linter rules for this file are coming from this laundry list of possible rules here: https://github.com/code-dot-org/code-dot-org/blob/0579d01daf5b014c54b1dcf41d3a2e328cfd9f34/.eslintrc.js if we start adding a lot more js in pegasus, we should consider getting more deliberate about what rules we want to enable in the public/js directories in pegasus, possibly including a predefined set like es5.

on the other hand, if most new js is going to be in react / in the apps/src directory then we are good. we get more specific in the apps directory where we specifically allow es6: https://github.com/code-dot-org/code-dot-org/blob/2d4125fc9b98f0d308240544579bb8f717e25be9/apps/.eslintrc.js#L14-L15 which we can do because we are running all that js through [babel](https://babeljs.io/).

food for thought 🥗 🧠

**Review by davidsbailey — APPROVED** [review]

**Comment by megcrenshaw** [comment]
> > Well done, Kelby! Love the callback solution for validation.
> > I think there's some follow-up work on the engineering side that I can take for now –– will loop you in on those conversations!
> 
> Woo hoo, thank you! Do you want to do another review since I made a few updates to address comments, or should I be ok to merge once the tests pass?

It's ready freddy! Thanks for diving in here.

File: pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js
Comment (kelbyhawn) [reply pegasus/sites.v3/code.org/public/js/admins_email_signup_form.js:28]: I think it would be good to add more js in general to pegasus to make some stuff have better UX and/or delight as we keep redesigning pages 😄

File: pegasus/sites.v3/code.org/views/admins_email_signup_form.html
Comment (megcrenshaw) [reply pegasus/sites.v3/code.org/views/admins_email_signup_form.html:44]: The long-awaited doc that I was going to write "tomorrow" is here https://docs.google.com/document/d/1i9xuC95p9tyY83c0iJMU9GWzNKrdeu69d8B3Ure9ui4/edit#

Comment (kelbyhawn) [reply pegasus/sites.v3/code.org/views/admins_email_signup_form.html:44]: Awesome, thanks for writing this up @megcrenshaw!

**Pull request merged**
